pingrid/impl.py

```diff
@@ -90,6 +79,89 @@ def error_fig(error_msg="error"):
 FuncInterp2d = Callable[[Iterable[np.ndarray]], np.ndarray]
 
 
+class ColorScale:
+    
+    def __init__(self, name, colors, scale=None):
+        self.name = name
+        self.colors = colors
+        if scale is None:
+            self.scale = list(np.arange(len(colors)))
+        else:
+           if (np.diff(scale) < 0).any():
+              raise Exception("scale must be monotically increasing")
+           elif len(colors) == len(scale):
+               self.scale = scale
+           else:
+               raise Exception("scale must be same length as colors")
+
+    def reversed(self, name=None):
+        if name is None:
+            name = self.name + "_r"
+        return ColorScale(name, self.colors[::-1], self.scale)
+
+    def rescaled(self, new_min, new_max):
+        cs_val = np.array(self.scale)
+        scale = (cs_val - cs_val[0]) * (new_max - new_min) / (cs_val[-1] - cs_val[0]) + new_min
+        return ColorScale(self.name, self.colors, scale)
+
+    def to_rgba_array(self, lutsize=256):
+        cs = self.rescaled(0, lutsize-1)
+        nc =  len(cs.scale)
```

Comment:
It would be clearer to call this variable `n_anchors`. The number of colors prior to quantization is potentially infinite, and the number of colors after quantization is lutsize. This variable represents neither of those.